src/libraries/System.Net.Sockets/tests/FunctionalTests/CreateSocketTests.cs

```diff
@@ -465,6 +465,46 @@ public void Ctor_SafeHandle_UnknownSocket_Success()
             }
         }
 
+
+        [DllImport("libc")]
+        private static unsafe extern int socketpair(int domain, int type, int protocol, int* ptr);
+
+        [DllImport("libc")]
+        private static extern int close(int fd);
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        public unsafe void Ctor_SafeHandle_SocketPair_Success()
+        {
+            // This is platform dependent but it seems like this is same on all supported platforms.
+            const int AF_UNIX = 1;
+            const int SOCK_STREAM = 1;
+            Span<int> ptr = stackalloc int[2];
```

Comment:
Newbie C# question: why use `fixed` and `Span`, and not something like ìnt fds[2]`?

Reply:
Because that's not valid C# :-)